# Description: use posix advisory file access flags only when available
#  They aren't on Debian/kFreBSD
# Author: Damyan Ivanov <dmn@debian.org>
# Forwarded: no
--- a/src/utilities/nbackup.cpp
+++ b/src/utilities/nbackup.cpp
@@ -449,12 +449,16 @@ void NBackup::open_database_scan()
 	if (dbase < 0)
 		b_error::raise(uSvc, "Error (%d) opening database file: %s", errno, dbname.c_str());
 
+#if defined POSIX_FADV_SEQUENTIAL
 	int rc = fb_fadvise(dbase, 0, 0, POSIX_FADV_SEQUENTIAL);
 	if (rc)
 		b_error::raise(uSvc, "Error (%d) in posix_fadvise(SEQUENTIAL) for %s", rc, dbname.c_str());
+#endif // defined(POSIX_FADV_SEQUENTIAL)
+#if defined(POSIX_FADV_NOREUSE)
 	rc = fb_fadvise(dbase, 0, 0, POSIX_FADV_NOREUSE);
 	if (rc)
 		b_error::raise(uSvc, "Error (%d) in posix_fadvise(NOREUSE) for %s", rc, dbname.c_str());
+#endif // defined(POSIX_FADV_NOREUSE)
 
 #endif // WIN_NT
 }
